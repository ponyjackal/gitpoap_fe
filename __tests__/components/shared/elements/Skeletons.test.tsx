import { render } from '@testing-library/react';
import 'jest-styled-components';
import {
  BaseSkeleton,
  POAPBadgeSkeleton,
  ProfileImageSkeleton,
  TextSkeleton,
} from '../../../../src/components/shared/elements';

describe('Skeletons', () => {
  describe('BaseSkeleton', () => {
    it('renders a BaseSkeleton', () => {
      const { container } = render(<BaseSkeleton />);
      const skeleton = container.firstChild;

      expect(skeleton).toBeInTheDocument();
      expect(skeleton).toMatchSnapshot();
    });
  });

  describe('POAPBadgeSkeleton', () => {
    it('renders a POAPBadgeSkeleton', () => {
      const { container } = render(<POAPBadgeSkeleton />);
      const skeleton = container.firstChild;

      expect(skeleton).toBeInTheDocument();
      expect(skeleton).toMatchSnapshot();
    });
  });

  describe('ProfileImageSkeleton', () => {
    it('renders a ProfileImageSkeleton', () => {
      const { container } = render(<ProfileImageSkeleton />);
      const skeleton = container.firstChild;

      expect(skeleton).toBeInTheDocument();
      expect(skeleton).toMatchSnapshot();
    });
  });

  describe('TextSkeleton', () => {
    it('renders a TextSkeleton', () => {
      const { container } = render(<TextSkeleton />);
      const skeleton = container.firstChild;

      expect(skeleton).toBeInTheDocument();
      expect(skeleton).toMatchSnapshot();
    });
  });
});
