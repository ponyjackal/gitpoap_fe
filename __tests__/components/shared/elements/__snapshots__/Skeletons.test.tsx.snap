// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeletons BaseSkeleton renders a BaseSkeleton 1`] = `
<div
  class="mantine-Skeleton-root mantine-Skeleton-visible mantine-1iz9qq"
/>
`;

exports[`Skeletons POAPBadgeSkeleton renders a POAPBadgeSkeleton 1`] = `
<div
  class="mantine-Skeleton-root mantine-Skeleton-visible mantine-1fykzg9"
/>
`;

exports[`Skeletons ProfileImageSkeleton renders a ProfileImageSkeleton 1`] = `
<div
  class="mantine-Skeleton-root mantine-Skeleton-visible mantine-r0vufu"
/>
`;

exports[`Skeletons TextSkeleton renders a TextSkeleton 1`] = `
<div
  class="mantine-Skeleton-root mantine-Skeleton-visible mantine-gnc43r"
/>
`;
